reposerver/repository/app_version.go

```diff
@@ -61,8 +61,16 @@ func getVersionFromFile(appPath, jsonPathExpression string) (*string, error) {
 	if err != nil {
 		return nil, err
 	}
-	appVersion, ok := versionValue.(string)
-	if !ok {
+
+	var appVersion string
+	var conversionSuccess bool
+	if versionArray, ok := versionValue.([]interface{}); ok && len(versionArray) > 0 {
```

Comment:
could you please move it to dedicated function that will return two values and cover it with primitive unit test. It is pretty sensitive code that may cause crash of app if something will went wrong